floodwatch/src/js/components/Main.js

```diff
@@ -107,18 +107,20 @@ export class Main extends Component {
 
     const navs = this.state.user ? SIGNED_IN_NAVS : SIGNED_OUT_NAVS;
 
+    const navElem = (window.location.pathname !== "/generate") ? <Navigation navs={navs} /> : null;
```

Comment:
Do you mean something like this, where it's not part of the Route that renders with Main?
```
      <Router history={history}>
        <Route path="/" component={Main}>
          <IndexRoute component={Landing} />
          ...
          <Route path="compare" component={Compare} onEnter={requireAuth} />
        </Route>
        <Route path="generate" component={Generate}/>
      </Router>
```

(AFAIK, we don't need anything from Main.js (checking for existence of the extension, the navs, etc.) to run in `/generate` anyway.)